src/niri.rs

```diff
@@ -4633,28 +4636,28 @@ impl Niri {
             let _ = tx.send(buf.clone());
 
             let mut image_path = None;
-
-            if let Some(path) = path {
-                debug!("saving screenshot to {path:?}");
-
-                if let Some(parent) = path.parent() {
-                    if let Err(err) = std::fs::create_dir(parent) {
-                        if err.kind() != std::io::ErrorKind::AlreadyExists {
-                            warn!("error creating screenshot directory: {err:?}");
+            if write_to_disk {
+                if let Some(path) = path {
```

Comment:
I think we want to print the debug message below in both cases, so something like: `if write_to_disk.then_some(path).flatten() {`

Reply:
Actually, even better idea: just do `path = None` if `!write_to_disk` at the top of this function, don't even call `make_screenshot_path()`